### PR TITLE
Improvements(UI): UI improvements for teams, users and admins pages.

### PR DESCRIPTION
**Pull request by aniketkatkar97**
### Describe your changes :

I worked on UI improvements for teams, users and admins pages.

#
### Type of change :

- [x] Bug fix
- [x] Improvement

#
### Frontend Preview (Screenshots) :
<p align="center">

- [x] Replaced email with roles column in user and admin details table
- [x] Converted teams and roles in user details table to links which will redirect user to respective entity
Screenshot 2022-09-02 at 6 23 48 PM
Screenshot 2022-09-02 at 6 23 56 PM
Screenshot 2022-09-02 at 6 24 06 PM

- [x] Teams page open group switch moved to manage menu.
Screenshot 2022-09-02 at 4 46 52 PM
Screenshot 2022-09-02 at 4 47 00 PM

- [x] Optimised API calls for teams page
- [x] Removed duplication of fields param in API call
- [x] Fixed bug for soft deleted team showing option for soft delete again
- [x] Fixed failing cypress tests
- [x] Minor bug fixes.

</p>

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers

Frontend: @open-metadata/ui

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/src/components/TeamDetails/TeamDetailsV1.tsx [hunk 1]
```diff
@@ -197,16 +199,19 @@ const TeamDetailsV1 = ({
...
-        dataIndex: 'email',
-        key: 'email',
-      },
       {
         title: 'Teams',
         dataIndex: 'teams',
         key: 'teams',
-        render: (teams: EntityReference[]) => getTeamsText(teams),
+        render: (teams: EntityReference[]) =>
+          getEntitiesText(teams, EntityType.TEAM),
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/components/TeamDetails/TeamDetailsV1.tsx:207]: As we have a details page for both teams and roles so can we make them clickable?

File: openmetadata-ui/src/main/resources/ui/src/components/TeamDetails/TeamDetailsV1.tsx [hunk 2]
```diff
@@ -483,6 +488,52 @@ const TeamDetailsV1 = ({
     return `Are you sure you want to ${text}`;
   };
 
+  const openGroupIcon = useMemo(() => {
+    if (currentTeam.isJoinable) {
+      return <SVGIcons alt="Delete" icon={Icons.OPEN_LOCK} />;
+    } else {
+      return <SVGIcons alt="Delete" icon={Icons.CLOSED_LOCK} />;
+    }
```
Comment (ShaileshParmar11) [inline openmetadata-ui/src/main/resources/ui/src/components/TeamDetails/TeamDetailsV1.tsx:496]:
```suggestion
  return <SVGIcons alt="Delete" icon={currentTeam.isJoinable ? Icons.OPEN_LOCK : Icons.CLOSED_LOCK} />
```

File: openmetadata-ui/src/main/resources/ui/src/components/TeamDetails/TeamDetailsV1.tsx (hunk 1 above)
Comment (Sachin-chaurasiya) [reply openmetadata-ui/src/main/resources/ui/src/components/TeamDetails/TeamDetailsV1.tsx:207]: Like we are doing for roles and policy listing page.
image

File: openmetadata-ui/src/main/resources/ui/src/components/TeamDetails/TeamDetailsV1.tsx [hunk 3]
```diff
@@ -483,6 +488,52 @@ const TeamDetailsV1 = ({
...
+            {openGroupIcon}
+            <div className="tw-text-left open-group" data-testid="open-group">
+              <Row className="tw-mb-1" justify="space-between">
+                <Col>
+                  <p className="tw-font-medium" data-testid="open-group-label">
+                    Open Group
+                  </p>
+                </Col>
+                <Col>
+                  <Switch
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/components/TeamDetails/TeamDetailsV1.tsx:516]: Here we are disabling the switch if the user does not have permission, so can we add the tooltip here?

Comment (aniketkatkar97) [reply openmetadata-ui/src/main/resources/ui/src/components/TeamDetails/TeamDetailsV1.tsx:516]: We are not disabling the switch, we are removing the manage dropdown itself. I will remove the disable prop as it is not necessary now.

File: openmetadata-ui/src/main/resources/ui/src/components/TeamDetails/TeamDetailsV1.tsx (hunk 1 above)
Comment (aniketkatkar97) [reply openmetadata-ui/src/main/resources/ui/src/components/TeamDetails/TeamDetailsV1.tsx:207]: Made the changes

File: openmetadata-ui/src/main/resources/ui/src/components/TeamDetails/TeamDetailsV1.tsx (hunk 2 above)
Comment (aniketkatkar97) [reply openmetadata-ui/src/main/resources/ui/src/components/TeamDetails/TeamDetailsV1.tsx:496]: Made the changes

**Comment by Sachin-chaurasiya** [comment]
@aniketkatkar97 , Can you merge the main branch?

**Review by ShaileshParmar11 — APPROVED** [review]

**Review by Sachin-chaurasiya — APPROVED** [review]

**Comment by Sachin-chaurasiya** [comment]
@aniketkatkar97 , Cypress test is failing for the team page.

**Review by Sachin-chaurasiya — APPROVED** [review]

**Pull request merged**